### ISSUE TITLE
Implement keyword redirects in the search results

### ISSUE THREAD

**Issue by tobi-or-not-tobi** [SAP/spartacus#4977]
The search configuration allows for keyword redirects, which means that a keyword can be mapped to a storefront URL, product, category or page. A classic example is to show/open a the cart page, when the user searches for "cart", "shopping cart", etc. Or open the order history page, when the user searches for "order". 

The search endpoint does already support this feature, when `keywordRedirectUrl` is added to the list of fields. 

Whenever there's a match with a `keywordRedirectUrl`, the user might choose to open the url. In case there's no other matches, we could automatically open the given link. 

This requires also a sample data change, as well as an alignment with the configured routes. 
 
The main reason to have keyword redirect configuration is based on the fact that there's a lack of content search. As long as this feature isn't provided by the backend, this is however beneficial for end users.

**Comment by tobi-or-not-tobi** [comment]
@vekterdude do we have already a good flow for the search redirects?

**Comment by bajo20003000** [comment]
It's not only about keywordRedirectUrl but also about Category, Product or Page, which are OOTB hybris features.

**Comment by vekterdude** [comment]
Minimum for me would be to support showing the destination in the autocomplete results. The larger feature I would tie to content results in with product results, but that is a longstanding missing piece from Acc and SPA.

Do you have  deadline to see something like that? or still just discussing?

**Comment by bajo20003000** [comment]
I assume it's a big gap. I expect it to work as in accelerator for the beginning.
Can you plan it and place in Spartacus roadmap?

**Comment by bajo20003000** [comment]
I assume it's a big gap. I expect it to work as in accelerator for the beginning.
Can you plan it and place in Spartacus roadmap?

**Comment by tobi-or-not-tobi** [comment]
Standard sample data doesn't include redirects for category pages, we'll move the following to the sample data. 

<details>
  <summary>ImpEx example category page redirects</summary>

```
$productCatalog=electronicsProductCatalog
$productCatalogVersion=catalogVersion(CatalogVersion.catalog(Catalog.id[default=$productCatalog]),CatalogVersion.version[default='Online'])[default=$productCatalog:Online]
$facetSearchConfigName=electronicsIndex

# Language
$lang=en

# Category Redirect (Must use INSERT_UPDATE here)
INSERT_UPDATE SolrCategoryRedirect;redirectItem(code,$productCatalogVersion)[unique=true];&redirectRefID
;brand_5;$productCatalog-redirectRefID-sony
;578;$productCatalog-redirectRefID-slr

INSERT_UPDATE SolrFacetSearchKeywordRedirect;facetSearchConfig(name)[unique=true,default=$facetSearchConfigName];language(isocode)[unique=true,default=$lang];keyword[unique=true];matchType(code)[unique=true];redirect(&redirectRefID);ignoreCase[default=true]
 ;;;"sony";EXACT;$productCatalog-redirectRefID-sony;
 ;;;"slr";EXACT;$productCatalog-redirectRefID-slr;


```
</details>

**Comment by tobi-or-not-tobi** [comment]
We've created an implementation for this which can be released with 2.0. This will replicate the "accelerator" behaviour, meaning:
- there's no UI that helps the user understand that we're redirecting to a real page rather than a search result page. While we could come up with various approaches, the backend API doesn't tell us much; only a redirect URL is resolved, no context, label, etc. 
- there's no content search involved in this story, that's really a different animal and should be separated from this ticket

We'll have feature parity with the accelerator for this feature. It behaves like this:
1. user types a search string
2. the suggestions and products will show up in the fly-out regardless of redirects
3. (only) if the use hits enter, and if there's a matching redirect, we redirect the user to the given (external) page.

The experience is obviously not optimised, but it's a start and it will give us feature parity with the accelerator.

**Comment by Platonn** [comment]
This feature has not been implemented in `develop` yet. There is only a WIP branch, quite outdated from develop: https://github.com/SAP/spartacus/compare/feature/GH-4977?expand=1
